Building master fails on arm

Hi all,

Thanks for your great work on the language. When building directly on arm (v7hf) or through zig-bootstrap (arm-linux-musleabihf), the compilation fails on what appear to be integer size issues:

```
./src/link/Plan9.zig:465:50: error: expected type 'usize', found 'u64'
        decl.link.plan9.got_index = self.got_len - 1;
                                                 ^
./src/link/Plan9.zig:465:50: note: unsigned 32-bit int cannot represent all possible unsigned 64-bit values
        decl.link.plan9.got_index = self.got_len - 1;
                                                 ^
./src/Module.zig:4159:9: note: referenced here
        try mod.comp.bin_file.allocateDeclIndexes(new_decl);
        ^
./src/Module.zig:4683:28: note: referenced here
        const array_decl = try mod.createAnonymousDeclFromDecl(decl, .{
                           ^
./src/Compilation.zig:1755:17: note: referenced here
                try module.populateTestFunctions();
                ^
./src/main.zig:2436:5: note: referenced here
    try comp.update();
    ^
./src/main.zig:2131:35: note: referenced here
    updateModule(gpa, comp, hook) catch |err| switch (err) {
                                  ^
./src/stage1.zig:50:43: note: referenced here
        stage2.mainArgs(gpa, arena, args) catch |err| fatal("{s}", .{@errorName(err)});

./src/Sema.zig:5623:64: error: expected type 'usize', found 'u64'                                                           
                const buf = try anon_decl.arena().alloc(Value, final_len);
                                                               ^
./src/Sema.zig:5623:64: note: unsigned 32-bit int cannot represent all possible unsigned 64-bit values
                const buf = try anon_decl.arena().alloc(Value, final_len);
                                                               ^
./src/Sema.zig:5623:56: note: referenced here
                const buf = try anon_decl.arena().alloc(Value, final_len);
                                                       ^
./src/Sema.zig:5698:60: error: expected type 'usize', found 'u64'
            const buf = try anon_decl.arena().alloc(Value, final_len);
                                                           ^
./src/Sema.zig:5698:60: note: unsigned 32-bit int cannot represent all possible unsigned 64-bit values
            const buf = try anon_decl.arena().alloc(Value, final_len);
                                                           ^
./src/Sema.zig:5698:52: note: referenced here
            const buf = try anon_decl.arena().alloc(Value, final_len);
                                                   ^
./src/value.zig:1619:34: error: expected type 'usize', found 'u64'
            lhs_bigint.limbs.len - (shift / (@sizeOf(std.math.big.Limb) * 8)),
                                 ^
./src/value.zig:1619:34: note: unsigned 32-bit int cannot represent all possible unsigned 64-bit values
            lhs_bigint.limbs.len - (shift / (@sizeOf(std.math.big.Limb) * 8)),
                                 ^
./src/value.zig:1617:42: note: referenced here
        const limbs = try allocator.alloc(
                                         ^
./src/Sema.zig:5502:25: note: referenced here
            const val = try lhs_val.shr(rhs_val, sema.arena);

./src/Sema.zig:6648:32: error: expected type 'Module.CompileError!u32', found '@typeInfo(@typeInfo(@TypeOf(Sema.safetyPanic)).Fn.return_ty
pe.?).ErrorUnion.error_set!u32'
        return sema.safetyPanic(block, src, .unreach);
                               ^
./src/Sema.zig:6648:32: note: error set '@typeInfo(@typeInfo(@TypeOf(Sema.safetyPanic)).Fn.return_type.?).ErrorUnion.error_set' cannot cas
t into error set 'Module.CompileError'
        return sema.safetyPanic(block, src, .unreach);
```

Thanks @g-w1 ! Looking at the above it seems that there are also issues in `Sema.zig` and `value.zig` as well as a potentially unrelated issue 

```
Sema.zig:6648:32: error: expected type 'Module.CompileError!u32', found '@typeInfo(@typeInfo(@TypeOf(Sema.safetyPanic)).Fn.return_ty
pe.?).ErrorUnion.error_set!u32'
        return sema.safetyPanic(block, src, .unreach);
```
Are these all related?

--- a/src/link/Plan9.zig
+++ b/src/link/Plan9.zig
@@ -36,7 +36,7 @@ hdr: aout.ExecHdr = undefined,
 
 entry_val: ?u64 = null,
 
-got_len: u64 = 0,
+got_len: usize = 0,
 
 const Bases = struct {
     text: u64,
@@ -243,7 +243,7 @@ pub fn flushModule(self: *Plan9, comp: *Compilation) !void {
     const hdr_slice: []u8 = hdr_buf[0..hdr_size];
     var foff = hdr_size;
     iovecs[0] = .{ .iov_base = hdr_slice.ptr, .iov_len = hdr_slice.len };
-    var iovecs_i: u64 = 1;
+    var iovecs_i: usize = 1;
     var text_i: u64 = 0;
     // text
     {

